Add D2Lang CRC-16 Calculate Checksum function

IAmTrial opened the pull request:
This adds a reimplementation of the CRC-16 calculate checksum function, D2Lang.Ordinal10005. The lookup table is replicated 100%, and each of the 4x4 matrix sets are described in comments. My own quickly written tests external to this repository confirms that the function outputs the exact same checksum as the vanilla 1.10f D2Lang.dll, when fed the same input.

Lectem commented:
It looks like you forgot to add the files to https://github.com/ThePhrozenKeep/D2MOO/blob/master/source/D2Lang/CMakeLists.txt

IAmTrial commented:
> It looks like you forgot to add the files to https://github.com/ThePhrozenKeep/D2MOO/blob/master/source/D2Lang/CMakeLists.txt

I have now added the files to the D2Lang CMakeLists.txt. In addition, I have also added the definitions to the def file.

IAmTrial commented:
I cloned the project again to figure out my CMake issues and got them sorted out. I have added changes to fix the compile error (missing D2Lang dependency on Fog) and to suppress narrowing warnings (int -> unsinged short).
